lib/services/android-project-service.ts

```diff
@@ -265,14 +248,64 @@ class AndroidProjectService implements IPlatformProjectService {
 
 			let libProjProp = path.join(libraryPath, "project.properties");
 			if (this.$fs.exists(libProjProp).wait()) {
-				this.updateProjectReferences(platformData.projectRoot, targetLibPath);
+				this.updateProjectReferences(platformData.projectRoot, targetLibPath).wait();
 			}
 		}).future<void>()();
 	}
-
+	
 	public getFrameworkFilesExtensions(): string[] {
 		return [".jar", ".dat"];
 	}
+	
+	public preparePluginNativeCode(pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let pluginPlatformsFolderPath = this.getPluginPlatformsFolderPath(pluginData);
+			
+			// Handle *.jars inside libs folder
+			let libsFolderPath = path.join(pluginPlatformsFolderPath, AndroidProjectService.LIBS_FOLDER_NAME);
+			if(this.$fs.exists(libsFolderPath).wait()) {
+				let libsFolderContents = this.$fs.readDirectory(libsFolderPath).wait();
+				_(libsFolderContents)
+					.filter(libsFolderItem => path.extname(libsFolderItem) === ".jar")
+					.map(jar => this.addLibrary(this.platformData, path.join(libsFolderPath, jar)).wait());
+			}
+			
+			// Handle android libraries
+			let androidLibraries = this.getAllAndroidLibrariesForPlugin(pluginData).wait();
+			_.each(androidLibraries, androidLibraryName => this.addLibrary(this.platformData, path.join(pluginPlatformsFolderPath, androidLibraryName)).wait());
+			
+		}).future<void>()();
+	}
+	
+	public removePluginNativeCode(pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let projectPropertiesManager = this.getProjectPropertiesManager(this.platformData.projectRoot);			
+			
+			let androidLibraries = this.getAllAndroidLibrariesForPlugin(pluginData).wait();			
+			_.each(androidLibraries, androidLibraryName => {
+				let libPath = path.join(this.$projectData.projectDir, "lib", this.platformData.normalizedPlatformName, androidLibraryName);
+				projectPropertiesManager.removeProjectReference(path.relative(this.platformData.projectRoot, libPath)).wait(); // TODO: Compose the reference path
+				this.$fs.deleteDirectory(libPath).wait();
+			});
+			
+		}).future<void>()();
+	}
+	
+	private getPluginPlatformsFolderPath(pluginData: IPluginData) {
+		return pluginData.pluginPlatformsFolderPath(AndroidProjectService.ANDROID_PLATFORM_NAME);
+	}
+	
+	private getAllAndroidLibrariesForPlugin(pluginData: IPluginData): IFuture<string[]> {
+		return (() => {
+			let pluginPlatformsFolderPath = this.getPluginPlatformsFolderPath(pluginData);
+			let platformsContents = this.$fs.readDirectory(pluginPlatformsFolderPath).wait();
```

Comment:
skip this in case the plugin does not have native library. Currently this code fails on `tns plugin remove tns-plugin` with
`Error: ENOENT, scandir 'd:\Work\nativescript-cli\scratch\app1004\node_modules\tns-plugin\platforms\android'`